deps/rabbit/src/rabbit_amqqueue_sup_sup.erl

```diff
@@ -57,15 +64,18 @@ find_for_vhost(VHost, Node) ->
 start_for_vhost(VHost) ->
     case rabbit_vhost_sup_sup:get_vhost_sup(VHost) of
         {ok, VHostSup} ->
-            supervisor2:start_child(
-              VHostSup,
-              {rabbit_amqqueue_sup_sup,
-               {rabbit_amqqueue_sup_sup, start_link, []},
-               transient, infinity, supervisor, [rabbit_amqqueue_sup_sup]});
+            StartMFA = {rabbit_amqqueue_sup_sup, start_link, []},
+            ChildSpec = #{id => rabbit_amqqueue_sup_sup,
+                          start => StartMFA,
+                          restart => transient,
+                          shutdown => infinity,
+                          type => supervisor,
+                          modules => [rabbit_amqqueue_sup_sup]},
+            supervisor:start_child(VHostSup, ChildSpec);
         %% we can get here if a vhost is added and removed concurrently
         %% e.g. some integration tests do it
         {error, {no_such_vhost, VHost}} ->
-            rabbit_log:error("Failed to start a queue process supervisor for vhost ~s: vhost no longer exists!",
+            rabbit_log:error("Failed to stop a queue process supervisor for vhost ~s: vhost no longer exists!",
```

Comment:
I don't think this change is correct, I will revert it. We are in start_for_vhost...